src/networking.c

```diff
@@ -0,0 +1,374 @@
+		   NETLINK_LINK_EN,
+		   config->net.bridge);
+
+	if ( !cc_oci_netlink_run(cmd_line) ) {
+		return false;
+	}
+
+	return true;
+}
+
+static gchar *
+get_address(const gint family, const void * const sin_addr)
+{
+	gchar addrBuf[INET6_ADDRSTRLEN];
+
+	inet_ntop(family, sin_addr, addrBuf, sizeof(addrBuf));
+
+	return(g_strdup(addrBuf));
+}
+
+static gchar *
+get_mac_address (const gchar * const ifname)
+{
+	struct ifreq ifr;
+	int fd = -1;
+	gchar *macaddr;
+	gchar *data;
+
+	fd = socket (AF_INET, SOCK_DGRAM, IPPROTO_IP);
+	if (fd < 0) {
+		g_critical("socket() failed with errno =  %d %s \n",
+			errno, strerror(errno));
+		return(g_strdup(""));
+	}
+
+	memset (&ifr, 0, sizeof (struct ifreq));
+	g_strlcpy (ifr.ifr_name, ifname, IFNAMSIZ);
+
+	if (ioctl (fd, SIOCGIFHWADDR, &ifr) < 0) {
+		g_critical("ioctl() failed with errno =  %d %s \n",
+			errno, strerror(errno));
+		macaddr = g_strdup("");
+		goto out;
+	}
+
+	if (ifr.ifr_hwaddr.sa_family != ARPHRD_ETHER) {
+		g_critical("invalid interface  %s \n", ifname);
+		macaddr = g_strdup("");
+		goto out;
+	}
+
+	data = ifr.ifr_hwaddr.sa_data;
+	macaddr = g_strdup_printf("%.2x:%.2x:%.2x:%.2x:%.2x:%.2x",
+				(unsigned char)data[0],
+				(unsigned char)data[1],
+				(unsigned char)data[2],
+				(unsigned char)data[3],
+				(unsigned char)data[4],
+				(unsigned char)data[5]);
+
+out:
+	close(fd);
+	return macaddr;
+}
+
+/*!
+ * Obtain the network configuration of a particular
+ * interface by querying the network namespace.
+ * Will be scanned from the namespace
+ * Ideally the OCI spec should be modified such that
+ * these parameters are sent to the runtime 
+ *
+ * \param[in] ifname \c interface name
+ * \param[in,out] config \ref cc_oci_config.
+ *
+ * \return \c true on success, else \c false.
+ */
+gboolean
+cc_oci_network_discover (const gchar * const ifname, 
+			  struct cc_oci_config * const config)
+{
+	struct ifaddrs *ifa = NULL;
+	struct ifaddrs *ifaddrs = NULL;
+	gint family;
+
+	if ((! config) || (! ifname)) {
+		return false;
+	}
+
+
+	if (getifaddrs(&ifaddrs) == -1) {
+		g_critical("getifaddrs() failed with errno =  %d %s \n",
+			errno, strerror(errno));
+		return false;
+	}
+
+	g_debug("Discovering container interfaces");
+	for (ifa = ifaddrs; ifa != NULL; ifa = ifa->ifa_next) {
+		if (ifa->ifa_addr == NULL) {
+			continue;
+		}
+
+		g_debug("	Interface := [%s]", ifa->ifa_name);
+		if (g_strcmp0(ifa->ifa_name, ifname) != 0) {
+			continue;
+		}
+
+
+		family = ifa->ifa_addr->sa_family;
+		if ((family != AF_INET) && (family != AF_INET6)) {
+			continue;
+		}
+
+		/* We have found at least one interface that matches */
+		config->net.ifname = g_strdup (ifname);
+
+		/* Assuming interfaces has both IPv4 and IPv6 addrs */
+		if ( family == AF_INET6 ) {
+			config->net.ipv6_address = get_address(family,
+			&((struct sockaddr_in *)ifa->ifa_addr)->sin_addr);
+
+			g_debug("Container IPv6 IP := [%s]", 
+				config->net.ipv6_address);
+
+			continue;
```

Comment:
yes, they are freed by cc_oci_config_free but you are not using it
let's do to your code an infinite loop
```
while(1) {    /* for (ifa = ifaddrs; ifa != NULL; ifa = ifa->ifa_next)  */
.....
    config->net.ifname = g_strdup (ifname); /* allocate memory on the heap */

    if ( family == AF_INET6 ) {    /* assuming that family= AF_INET6 */
        config->net.ipv6_address = get_address(family,
         &((struct sockaddr_in *)ifa->ifa_addr)->sin_addr);  /* allocate memory on the heap again*/

         /* you do not call to cc_oci_config_free before continue!!!*/
         /* you do not free ipv6_address*/
         continue;
    }
.....
```